cmd/shield/main.go

```diff
@@ -1485,11 +1486,23 @@ func main() {
 				return err
 			}
 
-			if err := RunJob(id, string(b)); err != nil {
+			var taskUUID string = ""
+			if taskUUID, err = RunJob(id, string(b)); err != nil {
 				return err
 			}
 
-			OK("Scheduled immediate run of job")
+			if taskUUID != "" && *opts.Raw {
+				RawJSON(map[string]interface{}{
+					"ok":        "Scheduled immediate run of job",
+					"task_uuid": taskUUID,
+				})
+			} else {
+				OK("Scheduled immediate run of job")
+				if taskUUID != "" {
+					ansi.Printf("To view task, type @B{shield show task %s}\n", taskUUID)
+				}
+			}
+
```

Comment:
The original issue was the compound conditional checking for "do we have a task UUID _and_ do they want raw output?"

If so, we were providing raw output; otherwise we were printing human-friendly output.

The complaint was that if the task UUID is "", the consequent of the conditional is skipped, and it reads like this: "well, task UUID was empty, so I guess print human-readable output.  Granted, `Ok` will make that JSON, but it would be JSON without the `task_uuid` attribute, which would be confusing.  Also, if someone opted to print some additional messaging (like `fmt.Printf("but we found no task UUID!\n")` it would break JSON parsers.

IIRC, @thomasmmitchell already fixed the code.